pkg/kubelet/kubelet_pods.go

```diff
@@ -452,7 +450,12 @@ func (kl *Kubelet) GeneratePodHostNameAndDomain(pod *v1.Pod) (string, string, er
 		if msgs := utilvalidation.IsDNS1123Label(pod.Spec.Subdomain); len(msgs) != 0 {
 			return "", "", fmt.Errorf("pod Subdomain %q is not a valid DNS label: %s", pod.Spec.Subdomain, strings.Join(msgs, ";"))
 		}
-		hostDomain = fmt.Sprintf("%s.%s.svc.%s", pod.Spec.Subdomain, pod.Namespace, clusterDomain)
+		hostDomain = fmt.Sprintf("%s.%s.svc", pod.Spec.Subdomain, pod.Namespace)
```

Comment:
can you please add a unit test for this method? Code change lgtm, but it will be best if it will be covered with the unit tests

Reply:
Sure! I just added the unit test, let me know if there is anything else missing.